src/gui/filedetails/FileDetailsWindow.qml

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+import QtQuick 2.15
+import QtQuick.Window 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+
+import com.nextcloud.desktopclient 1.0
+import Style 1.0
+
+ApplicationWindow {
```

Comment:
when closing the window, it is not being destroyed so each time you open a new dialog it will create an extra one leaking resources

Reply:
Each time we open the tray window we run a garbage collection that destroys existing file detail that are closed, so in this way we do not have memory leaks

Having said this it would also be more efficient to A) run this garbage collection process each time we create a new file details window and B) to not create a new file details dialog window if we already have a file details window existing for the relevant file

I will add these changes

Reply:
Okay, this is now done -- we perform the GC whenever we open the tray window AND whenever we open a new file details dialog (in the second case we reopen the existing dialog if it exists)